### ISSUE TITLE
dayOfYear in leap years

### ISSUE THREAD

**Issue by ChrisSeakane** [spencermountain/spacetime#382]
dayOfYear doesn't work if I want to set to 31st December in a leap year.

```
let s = spacetime()
s = s.year(2020)
s = s.dayOfYear(366);
console.log(s.json())
```

I get 30th December (date = 30, month = 11)

![image](https://user-images.githubusercontent.com/81533380/230719336-42d9f802-fae8-48e8-942f-fbe483f3351a.png)

Maybe I'm stupid/misunderstanding something 🤷

**Comment by spencermountain** [comment]
Oof! This does look like a bug. Am offline for the weekend, let me take a look then.
Thanks for the heads up

**Comment by ChrisSeakane** [comment]
Of course, no rush. Enjoy the Easter weekend!

**Comment by spencermountain** [comment]
fixed in 7.4.3
cheers
